compliance_checker/suite.py

```diff
@@ -723,7 +723,13 @@ def load_remote_dataset(self, ds_str):
         :param str ds_str: URL to the remote resource
         '''
 
-        if opendap.is_opendap(ds_str):
+        if erddap.is_tabledap(ds_str):
+            return Dataset(
```

Comment:
If this is slower than downloading the file and running the checks locally as you say, could something like a temporary file be an alternative faster solution (I did not run any tests)?

Downloading the file locally and removing it when done could be something like:

```python
@contextmanager
def tempnc(data: BinaryIO) -> Generator[str, None, None]:
    from tempfile import NamedTemporaryFile

    tmp = None
    try:
        tmp = NamedTemporaryFile(suffix=".nc", prefix="compliance-checker_")
        tmp.write(data)
        tmp.flush()
        yield tmp.name
    finally:
        if tmp is not None:
            tmp.close()
```

Reply:
Do we need some way to limit the amount of data requested?  The goal of the check is really to check the dataset structure, not so much the contents of the whole dataset.  I'm not really sure how this part of CC works, but is there a row limit or something that can be included in the ERDDAP request, if we're not doing that already?

Reply:
There are a few tests that requires the data/coordinates but a `.cdl` only test (metadata only) for online uses is something I've been wanting for a long time. However, do we have an ERDDAP response that returns only the metadata?

Reply:
That's true, need to check the data for some tests.  Not sure about this one though, mostly attribute-only, with some dimension checks like this one.  Possibly forgetting some things tho.  

There's the `.das` response, for example: http://erddap.sensors.ioos.us/erddap/tabledap/ssbn7-sun2wave-sun2w-sunset-n.das.  

This is what Bob pointed me to as most useful for checking dataset metadata.

Reply:
With the OPeNDAP protocol, dimension information is typically be found in the `DDS`, at the `.dds` format, and would look something like this:

```
...
    Float64 mask_v[eta_v = 290][xi_v = 332];
    Float64 mask_psi[eta_psi = 290][xi_psi = 331];
    Float32 zeta[time = 762][eta_rho = 291][xi_rho = 332];
    Float32 u[time = 762][s_rho = 20][eta_u = 291][xi_u = 331];
    Float32 v[time = 762][s_rho = 20][eta_v = 290][xi_v = 332];
...
```

However, a request to an ERDDAP source doesn't bring back the same structure:

```
dalton@ubuntu:~$ curl -L "http://erddap.sensors.ioos.us/erddap/tabledap/ssbn7-sun2wave-sun2w-sunset-n.dds"                                               
Dataset {
  Sequence {
    Float64 time;
    Float64 latitude;
    Float64 longitude;
    Float64 z;
    Float64 sea_water_velocity_to_direction;
    Int32 sea_water_velocity_to_direction_qc_agg;
    String sea_water_velocity_to_direction_qc_tests;
    Float64 sea_water_speed;
    Int32 sea_water_speed_qc_agg;
    String sea_water_speed_qc_tests;
    Float64 sea_water_temperature;
    Int32 sea_water_temperature_qc_agg;
    String sea_water_temperature_qc_tests;
    Float64 peak_wave_period;
    Int32 peak_wave_period_qc_agg;
    String peak_wave_period_qc_tests;
    Float64 sea_surface_wave_significant_height;
    Int32 sea_surface_wave_significant_height_qc_agg;
    String sea_surface_wave_significant_height_qc_tests;
    Float64 sea_surface_wave_from_direction;
    Int32 sea_surface_wave_from_direction_qc_agg;
    String sea_surface_wave_from_direction_qc_tests;
    String station;
  } s;
} s;
```

I'll dig around in the documentation to see if that's a possibility.

For the time being, the logic in this commit produces the desired results of requesting the `.ncCF` format.

Reply:
Ah, the `.ncCFHeader` might yield what we're looking for:

```
$ curl -L "http://erddap.sensors.ioos.us/erddap/tabledap/ssbn7-sun2wave-sun2w-sunset-n.ncCFHeader"
netcdf ssbn7-sun2wave-sun2w-sunset-n_e698_69e3_e309.nc {
  dimensions:
    timeseries = 1;
    profile = 127371;
    obs = 254742;
    station_strlen = 1;
    sea_water_velocity_to_direction_qc_tests_strlen = 1;
    sea_water_speed_qc_tests_strlen = 1;
    sea_water_temperature_qc_tests_strlen = 1;
    peak_wave_period_qc_tests_strlen = 1;
    sea_surface_wave_significant_height_qc_tests_strlen = 1;
    sea_surface_wave_from_direction_qc_tests_strlen = 1;
  variables:
    double latitude(timeseries=1);
      :_CoordinateAxisType = "Lat";
      :axis = "Y";
      :ioos_category = "Location";
      :long_name = "Latitude";
      :standard_name = "latitude";
      :units = "degrees_north";

    double longitude(timeseries=1);
      :_CoordinateAxisType = "Lon";
      :axis = "X";
      :ioos_category = "Location";
      :long_name = "Longitude";
      :standard_name = "longitude";
      :units = "degrees_east";
...
```

Reply:
Oh, yeah.  Duh, I've been staring at those forever, should have thought of that.  

For the `.das` response, the reason that may not show dimensions like other DAP servers is that the different ERDDAP output formats have different dimensionality, so it can't account for each of them.

Reply:
@ocefpaf: there's also the option to read a netCDF file into memory.  It should probably be preferred over writing to a tempfile when possible and when the netCDF-C library has been compiled with in-memory support.

Reply:
Indeed. But if we are aiming for a broader audience we should not rely on an option that may not be available, right?

PS: I believe that the conda-forge pkg is compiled with in-memory support but I'm not 100% sure. I'll check and fix it if not.

Reply:
@daltonkell where do we stand with merging this PR and resolving this dimension check issue?    Interested to include this in the next RC, if possible.   

Can we implement the dimension check, and maybe most or all of the IOOS checks, by requesting the `.ncCFHeader` response instead of the full `.ncCF` output?  For some ERDDAP datasets, if there aren't limits placed on the request, any of the `.nc`, `.ncCF`, `.ncCFMA`, or even `.csv` output types could end up requesting a lot of data that could contribute to poor performance.

That may be too big a change though.  If so, let's go with `.ncCF` and see what results are for next RC.

Reply:
I think including this in the next RC is appropriate, but we'll have to leave the `.ncCFHeader` request out for another edition. It's a great idea and perhaps we can get some good contributions for it, but it requires finding a way to "switch off" the checks which examine data (which would obviously fail).

I'm currently working on merging concepts from this PR and Ben's latest, #800, because his implements a useful abstraction for handling any remote netCDF resource.